### ISSUE TITLE
Send silent LFE signal?

### ISSUE THREAD

**Issue by mrmatteastwood** [kcat/openal-soft#798]
Hello, @kcat! I noticed something really weird about my setup the other day.

I have my Linux Mint 21 PC hooked up to my Denon AVR-X1400H 7.2 receiver via HDMI from my GTX 1080Ti grpahics card. In LM's sound settings, I have selected the 7.1 profile. When I play back 5.0 or 4.0 music files, i.e., content that doesn't have an LFE channel, my sound system's bass management doesn't reroute the low frequencies to the subwoofer.

So, for instance, if I play back the quadraphonic mix of Dark Side of the Moon, I get some decent bass from FL and FR, but the sub stays silent even though the sound system (Denon AVR) is configured to route frequencies below 80 Hz to the LFE.

Here's what baffles me: As soon as I add a silent (!!) LFE channel to these 4.0 files, making them 4.1 files, the subwoofer springs to life! I have no earthly idea what about sending a silent LFE channel prompts the Denon's bass management to kick in, but it does.

Now I'm wondering if there's any way to configure OpenAL to send a silent LFE as well. From what I understand, it only sends 6 channels (fronts, sides, rears) by default and, if I enable the front-stabilizer, it also sends a synthesized center.

How would I go about sending a silent LFE and tricking my sound system's bass management into working?

Thanks!
Matt.

**Comment by kcat** [comment]
If OpenAL Soft is outputting 5.1 or 7.1, it is sending a silent LFE signal. Once a context is set up, it's continuously sending a silent stream for all output channels, and mixes whatever sounds and effects are playing on top of that silence as needed.

Presumably you're using PipeWire, PulseAudio, or JACK, in which case it would be doing the same thing (except it may also put the device in a suspended state when nothing it playing). They don't dynamically reconfigure the channel configuration based on what's playing, so even if you're playing a 4.0 or 5.0 file, the server would be mixing that to, and playing as, 7.1, and simply send silence for any channels that don't have anything to play.

I don't know how you added the silent LFE channel to make it 4.1, or what you're using to play them, but maybe whatever's playing it is confused with the way that extra channel is presented in the file, and sending a non-silent signal to LFE.

**Comment by mrmatteastwood** [comment]
Thanks for responding so quickly and with this much detail. This is good to know! I inspected my sound setup further in the meantime and noticed that in fact, the sub response isn't actually the problem. The sub seems to work either way, and still there's more bass in the music when the FLAC files have 6 channels instead of 4 or 5. Anyway, OpenAL is clearly working just fine, then. Closing this issue.
